backend/app/src/web.rs

```
@@ -34,40 +33,40 @@ pub struct ApiDoc;
 #[derive(Clone)]
 struct AppState {
     flows_dir: flow::FlowsDir,
-    repo: Arc<Mutex<Option<git2_ox::Repository>>>,
+    git_actor: actors::git::GitActorAddr,
 }
 
 impl AppState {
     pub fn new(flows_dir: flow::FlowsDir) -> Self {
-        let repo = match flows_dir.path().parent() {
-            Some(p) => match git2_ox::Repository::try_open(p) {
-                Ok(repo) => Some(repo),
-                Err(_) => {
-                    log::warn!("Repository not found in '{}'", p.display());
-                    None
+        let git_actor = match flows_dir.path().parent() {
+            Some(p) => match crate::actors::git::GitActor::from_path(p) {
+                Ok(actor) => actor.spawn(),
+                Err(e) => {
+                    log::error!("Failed to create GitActor: {e}");
+                    panic!("Repository is required for GitActor");
```

maybe better to return the error and rename the method to try_new